### PR TITLE
chore(ts): first typescript first integration to communicator

### PR DESCRIPTION
**Pull request by barsukov**
# Summary

We make a typescript first integraiton to juno for simple communicator library.

# Changes Made

- New package for sharing typescript config
- New Eslint config that reflects Typescript 

# Related Issues
https://github.com/cloudoperators/juno/issues/119

# Testing Instructions

turbo lint

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

File: .github/licenserc.yaml [hunk 1]
```diff
@@ -14,7 +14,8 @@ header:
     - "apis/**"
     - "assets/**"
     - ".npmignore"
-    - "**/*.npmignore"
+    - ".npmignore"
+    - ".prettierrc"
```
Comment (ArtieReus) [inline .github/licenserc.yaml:18]: This extension is alredy added to the end of the list

File: packages/typescript-config/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# `typescript-config`
```
Comment (ArtieReus) [inline packages/typescript-config/README.md:1]: Do we need this file empty?

File: packages/communicator/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,279 @@
+/* eslint-disable */
+/* tslint:disable */
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
```
Comment (ArtieReus) [inline packages/communicator/src/index.ts:4]: No sure if the license heade will be found if it is not at the beginning. Can we add the eslint comments below?

**Review by ArtieReus — CHANGES_REQUESTED** [review]

File: packages/typescript-config/README.md (hunk 2 above)
Comment (barsukov) [reply packages/typescript-config/README.md:1]: I will write it.

File: .github/licenserc.yaml (hunk 1 above)
Comment (barsukov) [reply .github/licenserc.yaml:18]: Good catch

File: packages/communicator/src/index.ts (hunk 3 above)
Comment (barsukov) [reply packages/communicator/src/index.ts:4]: I will try to move it, this is the thing with automatic updates. I am not sure that should we do it as a workflow or just fail and force to do it like a lint task(do it locally and update it by your own...) lets discuss it.

**Review by ArtieReus — APPROVED** [review]
lgtm

**Pull request merged**